### PR TITLE
Fix/admin path serve

### PR DESCRIPTION
**Pull request by jozefcipa**
This PR fixes a bug where it tries to serve static assets instead of matching a **content type** URL endpoint when `admin.path` is set to `/` (with an intention to serve the UI on a different server)

So if my `config/environments/development/server.json` looks like this:
```
{
  "host": "localhost",
  "port": 1337,
  "proxy": {
    "enabled": false
  },
  "cron": {
    "enabled": false
  },
  "admin": {
    "path": "/",
    "autoOpen": false
  }
}
```
in the current state it loads a static content
image

whereas the right response should be a JSON API response
image

**Edit:** This PR adds `serveAdminPanel` config option to specify if it should serve static assets

### CODE REVIEW

**Comment by alexandrebodin** [comment]
Setting '/' as the path to disable the admin from being served doesn't really seem like a clear way of doing it for a user. We could just set the admin to false to disable it instead.

**Comment by jozefcipa** [comment]
I just followed the official documentation, but I agree, it would be definitely better. So do you think I should add `serveAdminPanel` to config or is there already another option for disabling it?

File: docs/3.0.0-beta.x/admin-panel/deploy.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ You might want to change the path to access to the administration panel. Here th
 }
 ```
 
-**You have to rebuild the administration panel to make this work.** [Build instructions](#build).
+**You have to rebuild the administration panel to make this work.** [Build instructions](https://strapi.io/documentation/3.0.0-beta.x/admin-panel/customization.html#build).
```
Comment (alexandrebodin) [inline docs/3.0.0-beta.x/admin-panel/deploy.md]: You should be using relative links with .md files. Vuepress builds the final urls itself

**Comment by alexandrebodin** [comment]
> I just followed the official documentation, but I agree, it would be definitely better. So do you think I should add `serveAdminPanel` to config or is there already another option for disabling it?

You could add support for setting

```json
{
  "admin": false
}
```

**Comment by alexandrebodin** [comment]
Thank you for your contribution 💯

**Pull request merged**